Modules/_datetimemodule.c

```diff
@@ -3250,6 +3257,29 @@ date_fromtimestamp(PyObject *cls, PyObject *obj)
     if (_PyTime_ObjectToTime_t(obj, &t, _PyTime_ROUND_FLOOR) == -1)
         return NULL;
 
+#ifdef MS_WINDOWS
+    if (t < 0) {
+        if (_PyTime_localtime(0, &tm) != 0)
+            return NULL;
+
+        int negate = 0;
+        PyObject *date = date_fromtimestamp(cls, _PyLong_GetZero());
+        if (date == NULL) {
+            return NULL;
+        }
+        PyObject *result = NULL;
+        PyObject *delta = PyObject_CallFunction((PyObject*)&DELTA_TYPE(NO_STATE), "iO", 0, obj);
+        if (delta == NULL) {
+            Py_DECREF(date);
+            return NULL;
+        }
```

Comment:
You can move negate and result definitions here. There is no need to initialize result to NULL.